File: LeetCodeProblemsGo/LeetCode1665MinimumInitialEnergyToFinishTasks/LeetCode1665MinimumInitialEnergyToFinishTasks.go
package LeetCode1665MinimumInitialEnergyToFinishTasks

import "sort"

func minimumEffort(tasks [][]int) int {
	sort.Slice(tasks, func(i, j int) bool {
		if tasks[i][1]-tasks[i][0] != tasks[j][1]-tasks[j][0] {
			return tasks[i][1]-tasks[i][0] > tasks[j][1]-tasks[j][0]
		}
		return tasks[i][0] > tasks[j][0]
	})
	current, delta, used := 0, 0, 0
	for _, task := range tasks {
		if current < task[1] {
			delta = task[1] - current
			current += delta
			used += delta
		}
		current -= task[0]

	}
	return used
}

File: LeetCodeProblemsJava/src/Problems/LeetCode1665MinimumInitialEnergyToFinishTasks.java
package Problems;

import java.util.Arrays;
import java.util.Comparator;

public class LeetCode1665MinimumInitialEnergyToFinishTasks {
    public int minimumEffort(int[][] tasks) {
        Arrays.sort(tasks,
                Comparator
                        .<int[]>comparingInt(a -> a[0] - a[1])
                        .thenComparing((a, b) -> Integer.compare(b[1], a[1]))
        );
        int used=0;
        int current=0;
        int delta=0;
        for( var task : tasks ) {
            delta=task[1]-current;
            if(delta>0){
                used+=delta;
                current+=delta;
            }
            current-=task[0];
        }
        return used;
    }
}
